lib/contract.ts

```diff
@@ -1235,7 +1266,7 @@ export default class Contract {
 		contract: Contract,
 		options: { types?: Set<string> } = {},
 	): boolean {
-		const conjuncts = reduce(
+		const conjuncts: Contract[] = reduce(
```

Comment:
This could be shortened to
```suggestion
		const conjuncts: Contract[] = contract.metadata.requirements.compiled.getAll().concat(
			contract.getChildren().flatMap((child) => child.metadata.requirements.compiled.getAll())
		)
```
and should result in slightly less allocations/work, alternatively it could be
```suggestion
		const conjuncts: Contract[] = [
			...contract.metadata.requirements.compiled.getAll(),
			...contract.getChildren().flatMap((child) => child.metadata.requirements.compiled.getAll()),
		]
```
which is often nicer but in this specific instance is a little clunky